pkg/services/ngalert/api/tooling/definitions/alertmanager.go

```diff
@@ -615,6 +616,40 @@ func (c *PostableUserConfig) validate() error {
 	return nil
 }
 
+// Decrypt returns a copy of the configuration struct with decrypted secure settings in receivers.
+func (c *PostableUserConfig) Decrypt(decryptFn func(payload []byte) ([]byte, error)) (*PostableUserConfig, error) {
+	// Copy the configuration struct.
+	rawCfg, err := json.Marshal(c)
+	if err != nil {
+		return nil, err
+	}
+
+	var newCfg PostableUserConfig
+	if err := json.Unmarshal(rawCfg, &newCfg); err != nil {
+		return nil, err
+	}
```

Comment:
I strongly think we should avoid serialising/deserializing to/from JSON - it's wasteful and typically very expensive.

I have noticed we have access to a library to do it for us (you're free to copy the function if you'd like to avoid the library use), then we should return `PostableUserConfig` not as a pointer to ensure we signal to the consumer that this is a copy.

```golang
	newCfg := deepcopy.Copy(&c)

	// Iterate through receivers and decrypt secure settings.
	for _, rcv := range newCfg.(PostableUserConfig).AlertmanagerConfig.Receivers {
		for _, gmr := range rcv.PostableGrafanaReceivers.GrafanaManagedReceivers {
			for k, v := range gmr.SecureSettings {
				decoded, err := base64.StdEncoding.DecodeString(v)
				if err != nil {
					return newCfg.(PostableUserConfig), fmt.Errorf("failed to decode value for key '%s': %w", k, err)
				}

				decrypted, err := decryptFn(decoded)
				if err != nil {
					return newCfg.(PostableUserConfig), fmt.Errorf("failed to decrypt value for key '%s': %w", k, err)
				}

				gmr.SecureSettings[k] = string(decrypted)
			}
		}
	}
	return newCfg.(PostableUserConfig), nil
````